diff --git a/pulp/src/wasm.rs b/pulp/src/wasm.rs
index d1b84dd..ee482bf 100644
--- a/pulp/src/wasm.rs
+++ b/pulp/src/wasm.rs
@@ -1932,7 +1932,7 @@ impl Simd for RelaxedSimd {
 				.v128_xor(cast!(f32x4(-0.0, 0.0, -0.0, 0.0)), cast!(bb),)
 		);
 
-		cast!(self.mul_add_e_f32s(self.mul_add_e_f32s(cast!(c), bb_sign, yx), aa, xy))
+		cast!(self.mul_add_e_f32s(aa, xy, self.mul_add_e_f32s(bb_sign, yx, cast!(c))))
 	}
 
 	#[inline(always)]
@@ -1946,7 +1946,7 @@ impl Simd for RelaxedSimd {
 
 		let bb_sign = cast!(self.simd128.v128_xor(cast!(f64x2(-0.0, 0.0)), cast!(bb)));
 
-		cast!(self.mul_add_e_f64s(self.mul_add_e_f64s(cast!(c), bb_sign, yx), aa, xy))
+		cast!(self.mul_add_e_f64s(aa, xy, self.mul_add_e_f64s(bb_sign, yx, cast!(c))))
 	}
 
 	#[inline(always)]
@@ -2011,7 +2011,7 @@ impl Simd for RelaxedSimd {
 				.v128_xor(cast!(f32x4(-0.0, 0.0, -0.0, 0.0)), cast!(bb),)
 		);
 
-		cast!(self.mul_add_e_f32s(self.mul_f32s(bb_sign, yx), aa, xy))
+		cast!(self.mul_add_e_f32s(aa, xy, self.mul_f32s(bb_sign, yx)))
 	}
 
 	#[inline(always)]
@@ -2039,7 +2039,7 @@ impl Simd for RelaxedSimd {
 
 		let bb_sign = cast!(self.simd128.v128_xor(cast!(f64x2(-0.0, 0.0)), cast!(bb)));
 
-		cast!(self.mul_add_e_f64s(self.mul_f64s(bb_sign, yx), aa, xy))
+		cast!(self.mul_add_e_f64s(aa, xy, self.mul_f64s(bb_sign, yx)))
 	}
 
 	#[inline(always)]
